crates/swc_ecma_parser/src/parser/expr/ops.rs

```diff
@@ -319,7 +319,7 @@ impl<I: Tokens> Parser<I> {
 
         if is!(self, "await") {
             let ctx = self.ctx();
-            if ctx.in_function && !ctx.in_async && ctx.strict {
+            if ctx.in_function && !ctx.in_async {
```

Comment:
`await` should be report as error in "module environment" not "strict mode".
And it is reserved word in async function as well.